Documentation/contributing/development/bgp_cplane.rst

```diff
@@ -0,0 +1,99 @@
+
+The following describes the role of each node:
+
+* ``router0`` is an `FRRouting (FRR)`_ router. It is pre-configured with minimal peering settings with server0 and server1.
+* ``server0`` and ``server1`` are ``nicolaka/netshoot`` containers that each share a network namespace with their own Kind node.
+* ``server2`` is a non-Cilium ``nicolaka/netshoot`` node useful for testing traffic connectivity from outside of the k8s cluster.
+
+.. _FRRouting (FRR): https://frrouting.org/
+
+Deploy Lab
+----------
+
+.. code-block:: shell-session
+
+   $ make kind-bgp-v4
+
+.. note::
+        Above example setups IPv4 single-stack environment. You can change ``v4`` part to ``v6`` or ``dual`` to setup IPv6 single-stack or dual-stack environment respectively. Same for all below examples.
+
+Install Cilium on the Lab
+-------------------------
+
+Install Cilium on the lab with your favorite way. The following example assumes you are modifying the source and want to build your own image. The minimal mandatory Helm values are provided in ``contrib/containerlab/bgp-cplane-dev-v4/values.yaml``. If needed, you can add Helm values to deploy BGP Control Plane with a different Cilium configuration.
+
+.. code-block:: shell-session
+
+   $ make docker-cilium-image docker-operator-generic-image
+   $ kind --name bgp-cplane-dev-v4 load docker-image quay.io/cilium/cilium:latest quay.io/cilium/operator-generic:latest
+   $ cilium install --chart-directory install/kubernetes/cilium -f contrib/containerlab/bgp-cplane-dev-v4/values.yaml --helm-set image.override="quay.io/cilium/cilium:latest" --helm-set image.pullPolicy=Never --helm-set operator.image.override="quay.io/cilium/operator-generic:latest" --helm-set operator.image.pullPolicy=Never
+
+Peering with Router
+-------------------
+
+Peer Cilium nodes with FRR by applying a CiliumBGPPeeringPolicy:
+
+.. code-block:: shell-session
+
+   $ make kind-bgp-v4-apply-policy
+
+.. note::
+        At this point, there are only minimal peering settings on the policy and no advertisement configuration present. You need to edit policies, for example, with ``kubectl edit bgpp`` to realize your desired settings. If you need to change the router side, you can edit FRRouting settings with ``docker exec -it clab-bgp-cplane-dev-v4-router0 vtysh``.
+
+Validating Peering Status
+-------------------------
+
+.. code-block:: shell-session
+
+   $ cilium bgp peers
+
+Validating Connectivity to Cilium Nodes from Non-Cilium Node
+------------------------------------------------------------
+
+The below example validates connectivity from server2 to server0 (10.0.1.2) and server1 (10.0.2.2). You should see the packet goes through router0 (10.0.3.1).
+
+.. code-block:: shell-session
+
+   $ docker exec -it clab-bgp-cplane-dev-v4-server2 mtr 10.0.1.2 -r
+   Start: 2023-10-04T02:24:58+0000
+   HOST: server2                     Loss%   Snt   Last   Avg  Best  Wrst StDev
+   1.|-- 10.0.3.1                   0.0%    10    0.0   0.1   0.0   0.1   0.0
```

Comment:
I'm not familiar with the expected output here, but is there some whitespace misalignment here? The data in the columns don't seem to line up with their associated headers. For instance, I would have expected:

```
$ docker exec -it clab-bgp-cplane-dev-v4-server2 mtr 10.0.2.2 -r
Start: 2023-10-04T02:25:58+0000
HOST: server2                     Loss%   Snt   Last   Avg  Best  Wrst StDev
1.|-- 10.0.3.1                     0.0%    10    0.1   0.1   0.0   0.1   0.0
2.|-- 10.0.2.2                     0.0%    10    0.1   0.1   0.0   0.2   0.0
```

The same goes for the table in the second command.

Sometimes weird stuff can happen when copying and pasting command output, so I just wanted to double check.

Reply:
Yes, as you said, something wrong happened when I copied and pasted the output. I fixed it.